catkit2/services/phasics_cam/phasics_cam.py

```diff
@@ -0,0 +1,286 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Oct 10 16:56:37 2024
+
+@author: lameier - using original code by Hari who worked with Phasics
+"""
+import PySID4x as sid
+import numpy as np
+from catkit2.testbed.service import Service
+import threading
+
+class IllegalArgumentError(ValueError):
+    pass
+
+class PhasicsCam(Service):
+    """
+    A service class for the Phasics camera.
+
+    A camera service with functionality for taking measurements,
+    starting and stopping live acquisitions, and managing an image buffer.
+
+    Parameters
+    ----------
+    Service : catkit2.testbed.service.Service
+        The base service class from which this class inherits.
+    """
+    def __init__(self):
+        """
+        Initializes the PhasicsCam object.
+
+        Sets up the instrument parameters, including exposure time, image dimensions,
+        and frame buffer size. Also initializes data streams and registers commands.
+
+        Attributes
+        ----------
+        exposure_time : int
+            The default exposure time in milliseconds.
+        image_height : int
+            The height of the images in pixels.
+        image_width : int
+            The width of the images in pixels.
+        num_frames_in_buffer : int
+            The number of frames to maintain in the data buffer.
+        images : catkit2.testbed.service.DataStream
+            A data stream to manage phase map image frames.
+        intensity : catkit2.testbed.service.DataStream
+            A data stream to manage intensity image frames.
+        """
+        super().__init__('phasics_cam')
+        # in dummy mode fake data is generated by API or device
+        SID4Session = self.initialize_sid4_session(mode=self.config.get('mode', 'normal'))
+        self.exposure_time = self.config.get('exposure_time', 504)
+
+        self.num_frames_in_buffer = self.config.get('buffer_frames', 32)
+
+        # Load the mask file and initialize the camera
+        maskname = self.config.get('mask')
+        self.log.info(f'loading mask {maskname}')
+        SID4Session.change_mask(filename=maskname)
+        SID4Session.Camera.load()
+        SID4Session.Camera.set_attribute('Exposure', self.exposure_time)
+        SID4Session.modify_user_profile(wavelength=self.config.get('wavelength'))
+        SID4Session.Zernike.open()
+        poly_order = 9
+        SID4Session.Zernike.update_projection(poly_order, source='user profile')
+
+        self.session = SID4Session
+
+        # take an image to determine the size of the frames based on the mask provided
+        self.log.info('getting test image')
+        self.set_exposure_time(self.exposure_time)
+        test_image = self.take_measurement()
```

Comment:
I don't like having to take a measurement to know the size of the output images. Note: you can create a datastream with the shape of the largest image that you'll take (ie. it doesn't need to be the exact size) and then reshape the datastream in the `open()` based on a temp measurement. I'd put the `max_width` and `max_height` in the config.